### DISCUSSION TITLE
Catalog overview images have to much alias

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#3921] [Ideas]
Since v19 I always thought the images in the catelog view where to fuzzy. The image on the product page is sharp. But this overview is not to nice. I think the defaults could be set differently. Maybe bigger image, maybe higher resize quality. I think the current sizes are still from times where there was less bandwidth. 
Maybe the frontend could have something you can select to choose what rendering optimization should be used.

catelog
![catelog](https://github.com/OpenMage/magento-lts/assets/13450219/d8b08d8f-c250-4cd0-8f3e-90d0bafe83fd)

product
![product](https://github.com/OpenMage/magento-lts/assets/13450219/cf0993ab-7f99-41b8-92e4-cb11cabc7d32)

original
![original](https://github.com/OpenMage/magento-lts/assets/13450219/95abc207-9c8f-44b5-a408-832fbb914a76)

**Comment by fballiano** [comment]
https://github.com/OpenMage/magento-lts/pull/3518

best way that I use for my customers is installing https://github.com/colinmollenhour/Perfect_Watermarks

**Reply by f1-outsourcing** [reply]
Yes I think there are quite a few still available. It is more that I think it would be nice having this a bit better in a default setup. Maybe I will do pull request on this some time. Before I will go live, I will check this. This one you are suggesting uses external binaries of imagemagick. I am not allowing this external executing of binaries.

**Reply by fballiano** [reply]
it doesn't use external binaries, it uses the imagic php extension, if I'm not mistaken.

AFAIK it's GD2 that makes the blurry images and it's not because of a "setting" we use, at least at the time that I had this problem I couldn't find anything like that.

**Comment by pauldpauld** [comment]
IIRC I had this issue in Magento 1.9

In app/code/core/Mage/Catalog/Model/Product/Image.php (copied to local obviously...)
I changed this from 60 to 90
 **protected $_quality = 90;** 

Edit: Sorry, the below is for m2
There's also an extension to add this to the admin area;
https://github.com/superdav42/magento-image-quality
I am not on the latest version of Openmage so this might be different....

**Reply by fballiano** [reply]
in OpenMage the $_quality is already set to 90.

for my customers I use the perfectwatermark module with uses ImageMagik (as the M2 module you linked) and imagemagik works fine for resizing images ;-)

**Reply by luigifab** [reply]
If you like ImageMagick, take a look at [imagetragick.com](https://imagetragick.com/) :)

**Reply by f1-outsourcing** [reply]
I am only advocating for a moderate fix that makes the images more acceptable. I still block exec of binaries (lots in disable_functions) so executing convert will not even work. This 'moderate' fix should be default available in any hosting environment.

**Reply by f1-outsourcing** [reply]
> in OpenMage the $_quality is already set to 90.
> 
> for my customers I use the perfectwatermark module with uses ImageMagik (as the M2 module you linked) and imagemagik works fine for resizing images ;-)

This imagemagick is using exec not? To execute external function?

**Reply by f1-outsourcing** [reply]
@fballiano 

Do you know what is causing this behaviour? Because I never had any issues with default php setups resizing images. 

I just browsed some code of what I am using and it seems it just uses this GD. I have never ever been annoyed by an upload. I really think this is just some issue in the magento code.

```
            $image = new \Phalcon\Image\Adapter\GD("{$filePath}/{$fileName}");
            $image->resize($width, $height);
            $path = "{$filePath}/{$prefix}_{$fileName}";
            $image->save($path, $quality);
```

I have the impression this 'bad' quality is from a smaller image being displayed bigger by the browser. Because when I click open image in new tab I see there a smaller image. 

I think the size of the default catelog image should just set be higher, here I currently have 210x210

**Reply by f1-outsourcing** [reply]
changing it here app/code/core/Mage/Catalog/etc/config.xml
<small_width>500</small_width>

does not seem to have an effect.

**Reply by f1-outsourcing** [reply]
Ok I have learned something. Looks like these images are generated on the fly, so you do not need to upload them to test, you can just delete the stuff in /media/catalog/product/cache/
Looks also that this resizing is handled by eg. the list.phtml if you change the value there the image is created in cache from the original and looks good.
I assume the css will fix/resize the images bigger than 210 correctly.

So I guess the default values in these phtml having 210 could be changed maybe to something like 420?
